pygatt/backends/bluez/bluez.py

```diff
@@ -0,0 +1,304 @@
+        """
+        matches = []
+        for path, ifaces in self.get_managed_objects().iteritems():
+            if not path.startswith(base_path):
+                continue
+            if interface not in ifaces:
+                continue
+            i = ifaces.get(interface)
+            match = True
+            for p, v in props.iteritems():
+                if (p not in i) or (i[p] != v):
+                    match = False
+                    break
+            if match:
+                matches.append(self.object_by_path(path))
+        return matches
+
+    def get_managed_objects(self):
+        obj_manager = self.object_by_path('/',
+                interface=self.DBUS_OBJECT_MANAGER_INTERFACE)
+        return obj_manager.GetManagedObjects()
+
+
+class BluezBackend(BLEBackend):
+    """
+    Backend to pygatt that uses BlueZ's dbus interface on the system bus.
+
+    Works with bluez >= 5.42, prior 5.x versions may work when the experimental
+    interfaces are enabled by starting the daemon with the `-E' flag.
+    """
+
+    def __init__(self, hci_device=None, scan_filter={}, main_loop=None):
+        """
+        Initialize the Bluez backend
+
+        hci_device -- When set, use this adapter. Use any adapter otherwise
+        scan_filter -- a dict of filter options. Supported Keys and value types:
+                        UUIDs [str], RSSI int (threshold),
+                        Pathloss int (threshold), Transport str (le|bredr|auto)
+        main_loop -- an external GLib.MainLoop already running, default None is
+                     used to run one internally
+        """
+        if main_loop:
+            self._main_loop = main_loop
+            self._main_loop_thread = None
+        else:
+            self._main_loop = GLib.MainLoop()
+            self._main_loop_thread = Thread(target=self._main_loop.run)
+        self._subscriptions = []
+        self._bus = DBusHelper(SystemBus())
+        try:
+            self._adapter = self.find_adapter(hci_device)
+        except BLEError as e:
+            if hci_device:
+                raise BLEError("Bluetooth adapter `{}' not found".format(
+                        hci_device))
+            raise e
+        if not self._adapter.Powered:
+            raise BLEError("Bluetooth adapter `{}' not powered".format(
+                    hci_device))
+
+        self._scan_filter = scan_filter
+        self._discovered_devices = {}
+        self._connected_devices = {}
+
+    def find_adapter(self, pattern=None):
+        return self.find_adapter_in_objects(self._bus.get_managed_objects(),
+                                            pattern)
+
+    def find_adapter_in_objects(self, objects, pattern=None):
+        for path, ifaces in objects.iteritems():
+            adapter = ifaces.get(DBusHelper.ADAPTER_INTERFACE)
+            if adapter is None:
+                continue
+
+            try:
+                if (not pattern and adapter.Powered) or \
```

Comment:
I've found that (at least under Python 2.7), `adapter.Powered` is not evaluated to `True` and no devices are returned, even when the device is powered. After changing to `adapter['Powered']` the first powered adapter is returned properly.